src/components/CategorySelector.tsx

```diff
@@ -0,0 +1,273 @@
+          )}
+          onChangeType={onCategoryClick}
+        />
+        <RegistrationTypesRow
+          mainType={PublicationType.Presentation}
+          registrationTypes={filterRegistrationTypes(
+            Object.values(PresentationType).map((registrationType) => ({
+              value: registrationType,
+              text: t(`registration.publication_types.${registrationType}`),
+              selected: selectedCategories.includes(registrationType),
+            }))
+          )}
+          onChangeType={onCategoryClick}
+        />
+        <RegistrationTypesRow
+          mainType={PublicationType.Artistic}
+          registrationTypes={filterRegistrationTypes(
+            Object.values(ArtisticType).map((registrationType) => ({
+              value: registrationType,
+              text: t(`registration.publication_types.${registrationType}`),
+              selected: selectedCategories.includes(registrationType),
+            }))
+          )}
+          onChangeType={onCategoryClick}
+        />
+        <RegistrationTypesRow
+          mainType={PublicationType.MediaContribution}
+          registrationTypes={filterRegistrationTypes(
+            Object.values(MediaType).map((registrationType) => ({
+              value: registrationType,
+              text: t(`registration.publication_types.${registrationType}`),
+              selected: selectedCategories.includes(registrationType),
+            }))
+          )}
+          onChangeType={onCategoryClick}
+        />
+        <RegistrationTypesRow
+          mainType={PublicationType.ResearchData}
+          registrationTypes={filterRegistrationTypes(
+            Object.values(ResearchDataType).map((registrationType) => ({
+              value: registrationType,
+              text: t(`registration.publication_types.${registrationType}`),
+              selected: selectedCategories.includes(registrationType),
+            }))
+          )}
+          onChangeType={onCategoryClick}
+        />
+        <RegistrationTypesRow
+          mainType={PublicationType.ExhibitionContent}
+          registrationTypes={filterRegistrationTypes(
+            Object.values(ExhibitionContentType).map((registrationType) => ({
+              value: registrationType,
+              text: t(`registration.publication_types.${registrationType}`),
+              selected: selectedCategories.includes(registrationType),
+            }))
+          )}
+          onChangeType={onCategoryClick}
+        />
+        <RegistrationTypesRow
+          mainType={PublicationType.GeographicalContent}
+          registrationTypes={filterRegistrationTypes(
+            Object.values(OtherRegistrationType).map((registrationType) => ({
+              value: registrationType,
+              text: t(`registration.publication_types.${registrationType}`),
+              selected: selectedCategories.includes(registrationType),
+            }))
+          )}
+          onChangeType={onCategoryClick}
+        />
```

Comment:
Could this be a loop?

Reply:
hadde vært mye finere å lese da! Men må i så fall lage en array som definerer `mainType` og `registrationTypes` da. Vurderte det som at vinninga gikk opp i spinninga, men når jeg tenker over det nå så heller jeg kanskje mer om at det forenkler mer enn det kompliserer læll ja 🤔

Reply:
Syns det ble mye bedre med `map` ja, good spot!
Faktisk bittelitt mer effektivt også, siden vi slipper å hente ut `Object.values(JournalType)` etc for alle rader på hver render :D